Added CL options and functionality

evertdespiegeleer opened the pull request:
Thanks for the script! Something that bugged me is that when I'm working and capturing my work, a quick move to another app that shouldn't be captured was captured as well. I added the --app command line option to fix this.
I also added the --period option, to be able to change the time between captures easily.

evertdespiegeleer committed:
Added --app and --period CL option

With the --period option, the amount of seconds between consecutive captures can be specified. With the --app option, the screen will only be captured with a specified app in focus.

evertdespiegeleer committed:
Update README.md for new command line options

evertdespiegeleer committed:
🐛 Fixed a bug that would crash the script on non-macOS machines

potomak commented on README.md:
```diff
@@ -1,3 +1,4 @@
+
```
Nit: no need for a new line here.

potomak commented on README.md:
```diff
@@ -48,4 +55,8 @@ $ ./clean_up.sh
 
-[http://labs.laan.com/wp/2011/01/how-to-make-time-lapse-screencaptures-of-your-design-work-for-free-mac/](http://labs.laan.com/wp/2011/01/how-to-make-time-lapse-screencaptures-of-your-design-work-for-free-mac/)
\ No newline at end of file
+[http://labs.laan.com/wp/2011/01/how-to-make-time-lapse-screencaptures-of-your-design-work-for-free-mac/](http://labs.laan.com/wp/2011/01/how-to-make-time-lapse-screencaptures-of-your-design-work-for-free-mac/)# Screen capture time-lapse
+
+A simple script to make time-lapse videos of your work.
+
+Using it is as simple as following those two steps.
```
I don't think this part needs to be changed.

potomak commented on capture.sh:
```diff
@@ -2,6 +2,15 @@
 
 SCREENCAPTURE_CMD='unknown'
 
+#Parse the command line arguments
```
Nit: space after comment

```suggestion
# Parse the command line arguments
```

potomak commented on capture.sh:
```diff
@@ -11,9 +20,28 @@ else
   exit 1
 fi
 
+PERIODI=4 #default to 4 seconds between captures. Can be changed with --period command line option
```
Why did you decide to call this variable `PERIODI` instead of `PERIOD`?

evertdespiegeleer replied:
`PERIOD` indeed is a better variable name. I initially changed the script without the intent to start a PL. I had a small issue with `PERIOD`, so I changed it to `PERIODI` and forgot about it. It's `PERIOD` again without any problems now.

potomak commented on capture.sh:
```diff
@@ -11,9 +20,28 @@ else
   exit 1
 fi
 
+PERIODI=4 #default to 4 seconds between captures. Can be changed with --period command line option
```
Nit: space after comment

```suggestion
PERIODI=4 # default to 4 seconds between captures. Can be changed with --period command line option
```

potomak commented on capture.sh:
```diff
@@ -11,9 +20,28 @@ else
 fi
 
+PERIODI=4 #default to 4 seconds between captures. Can be changed with --period command line option
+if [ ! -z "${period}" ]
+    then
+      PERIODI="$period"
+fi
```
Do you think we need some kind of validation for this value?

evertdespiegeleer replied:
I added evaluation for both -app and --period. Good idea!

potomak commented on capture.sh:
```diff
@@ -11,9 +20,28 @@ else
+if [ ! -z "${app}" ] && test "$(uname)" != 'Darwin'
+    then
+      echo 'The --app option is currently only available on macOS.'
+      exit 0
```
Zero means success, what do you think about returning a non-zero value if `--app` is set for non supported OSes?

evertdespiegeleer replied:
Changed it!

potomak commented on capture.sh:
```diff
@@ -11,9 +20,28 @@ else
+      then
+        APPINFOCUS=`osascript -e 'tell application "System Events"' -e 'set frontApp to name of first application process whose frontmost is true' -e 'end tell'`
+  fi
+  if [ "${APPINFOCUS}" == "${app}" ] || [ -z "${app}" ] #if --app command line option is provided, only capture when the specified app is in focus
```
Nit: space after comment

```suggestion
  if [ "${APPINFOCUS}" == "${app}" ] || [ -z "${app}" ] # if --app command line option is provided, only capture when the specified app is in focus
```

potomak reviewed (changes requested):
Thanks for your contribution @evertdespiegeleer!

The code LGTM.

I left a couple of comments, mostly nits.

Before I merge the PR can you reply to my comments and fix the format of your code to be consistent with the rest of the script (2 spaces indentation, `if ...; then` on a single line, etc.) please?

evertdespiegeleer committed:
Provided the requested changes

Checks for both the --period and the --app option were added. The code format was made more harmonised with @potomak's original style.

evertdespiegeleer committed:
Update README.md

evertdespiegeleer committed:
Update README.md

evertdespiegeleer reviewed (commented):
Thanks for your reaction.
Excuse me for the quite clumsy initial formatting. I barely ever code plain SH, so I don't really have a style for it.
I took into account your requests and suggestions. I added checks for the --app and --period CL options and harmonised the formatting.

potomak commented on capture.sh:
```diff
@@ -11,9 +20,33 @@ else
-# screen captures
+PERIOD=4 # Default to 4 seconds between captures. Can be changed with --period command line option
+if [ ! -z "${period}" ]; then
+  if ! [ "$period" -eq "$period" ] 2>/dev/null || [ "$period" -eq "0" ]; then # Evaluate whether $period is a number and whether it is > 0
```
I think the check should use `-gt "0"` instead of `-eq "0"`. I also thing you don't need to check that `$period` is an integer twice, if `-gt` fails to validate the input that's enough.

```suggestion
  if ! [ "$period" -gt "0" ] 2>/dev/null; then # Evaluate whether $period is a number and whether it is > 0
```

potomak commented on capture.sh:
```diff
@@ -2,6 +2,15 @@
 
 SCREENCAPTURE_CMD='unknown'
 
+# Parse the command line arguments
+while [ $# -gt 0 ]; do
+    if [[ $1 == *"--"* ]]; then
+      v="${1/--/}"
+      declare $v="$2"
+    fi
+  shift
+done
```
Sorry for being pedantic, but I think that `if` and `shift` should be at the same level:

```suggestion
# Parse the command line arguments
while [ $# -gt 0 ]; do
  if [[ $1 == *"--"* ]]; then
    v="${1/--/}"
    declare $v="$2"
  fi
  shift
done
```

potomak commented on README.md:
```diff
@@ -21,7 +21,13 @@ $ ./capture.sh
 When you want to make a break just stop the process using `ctrl-c` and restart when you're done.
 
-Default capture rate is one screenshot every 4 seconds.
+Default capture rate is one screenshot every 4 seconds. This can however be adjusted with the --period command line option.
```
Nit: options format

```suggestion
Default capture rate is one screenshot every 4 seconds. This can however be adjusted with the `--period` command line option.
```

potomak commented on README.md:
```diff
@@ -21,7 +21,13 @@ $ ./capture.sh
-Default capture rate is one screenshot every 4 seconds.
+Default capture rate is one screenshot every 4 seconds. This can however be adjusted with the --period command line option.
+
+By default, the screen will be captured regardless of which app you have in focus. When you switch from your Fusion360 session to Music to choose another song, this will be captured. This behaviour can be adjusted with the --app command line option. By providing the name of an app, screen captures will only be made when that specific app is in focus. An example with all options is shown bellow. T ...
```
Nit: options format

```suggestion
By default, the screen will be captured regardless of which app you have in focus. When you switch from your Fusion360 session to Music to choose another song, this will be captured. This behaviour can be adjusted with the `--app` command line option. By providing the name of an app, screen captures will only be made when that specific app is in focus. An example with all options is shown bellow. The `--app` option is only available on macOS.
```

potomak reviewed (changes requested):
Thanks for updating the PR.

I think there is an issue with the period value validation I would like for you to check.

potomak commented:
@evertdespiegeleer I hope you'll have the time to make the changes requested otherwise I can work on them. Let me know, thanks again for collaborating.

potomak commented:
@evertdespiegeleer if you want I can make the last changes before merging, just allow me to edit please, see https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork
